s3-source-connector/src/main/java/io/aiven/kafka/connect/s3/source/utils/RecordProcessor.java

```diff
@@ -43,54 +41,52 @@ public final class RecordProcessor {
     private RecordProcessor() {
 
     }
-    public static List<SourceRecord> processRecords(final Iterator<List<AivenS3SourceRecord>> sourceRecordIterator,
+
+    public static List<SourceRecord> processRecords(final Iterator<AivenS3SourceRecord> sourceRecordIterator,
             final List<SourceRecord> results, final S3SourceConfig s3SourceConfig,
             final Optional<Converter> keyConverter, final Converter valueConverter,
-            final AtomicBoolean connectorStopped, final OutputWriter outputWriter, final Set<String> failedObjectKeys,
+            final AtomicBoolean connectorStopped, final Transformer transformer, final FileReader fileReader,
             final OffsetManager offsetManager) {
 
         final Map<String, String> conversionConfig = new HashMap<>();
         final int maxPollRecords = s3SourceConfig.getInt(S3SourceConfig.MAX_POLL_RECORDS);
 
         for (int i = 0; sourceRecordIterator.hasNext() && i < maxPollRecords && !connectorStopped.get(); i++) {
-            final List<AivenS3SourceRecord> recordList = sourceRecordIterator.next();
-            final List<SourceRecord> sourceRecords = createSourceRecords(recordList, s3SourceConfig, keyConverter,
-                    valueConverter, conversionConfig, outputWriter, failedObjectKeys, offsetManager);
-            results.addAll(sourceRecords);
+            final AivenS3SourceRecord aivenS3SourceRecord = sourceRecordIterator.next();
+            if (aivenS3SourceRecord != null) {
+                final SourceRecord sourceRecord = createSourceRecord(aivenS3SourceRecord, s3SourceConfig, keyConverter,
+                        valueConverter, conversionConfig, transformer, fileReader, offsetManager);
+                results.add(sourceRecord);
+            }
         }
 
         LOGGER.info("Number of records sent {}", results.size());
```

Comment:
Before release I think we should change this to debug, right now super useful for testing.